File: clients/csharp/Bots/PlayState.cs
using System.Text.Json.Serialization;

namespace CsClient.Bots
{
    /// <summary>
    /// Antwortdata vom Server bzgl. game data
    /// </summary>
    public class PlayState
    {
        [JsonPropertyName("bot")] public string Bot { get; set; }

        [JsonPropertyName("coin_id")] public int CoinId { get; set; }

        [JsonPropertyName("round")] public int Round { get; set; }

        [JsonPropertyName("bombs")] public List<Bomb> Bombs { get; set; }

        [JsonPropertyName("board")] public List<List<int>> Board { get; set; }

        public PlayState()
        {
        }

        public PlayState(string bot, int coinId, int round, List<Bomb> bombs, List<List<int>> board)
        {
            Bot = bot;
            CoinId = coinId;
            Round = round;
            Bombs = bombs;
            Board = board;
        }
    }
}

File: clients/go/internal/model/playstate.go
package model

// Bomb represents the bomb's attributes within the game.
type Bomb struct {
	Row             int `json:"row"`               // row of the bomb
	Col             int `json:"col"`               // column of the bomb
	ExplodedInRound int `json:"exploded_in_round"` // round when the bomb explodes
}

// PlayState represents the state of the game at any given moment.
type PlayState struct {
	Bot    string  `json:"bot"`     // name of the bot
	CoinId int     `json:"coin_id"` // bot's coin ID
	Round  int     `json:"round"`   // Current round of the match
	Bombs  []Bomb  `json:"bombs"`   // List of bombs in the game
	Board  [][]int `json:"board"`   // current representation of the game board
}
